zUtilities/QuickSave.cpp

```diff
@@ -279,12 +279,12 @@ namespace GOTHIC_ENGINE {
   }
 
   void QuickSave::MenuLoop() {
-    if ( !Options::QuickSaveMode || zCMenu::inGameMenu || isLoading || !zinput->KeyToggled( Options::KeyQuickLoad ) ) return;
+    if ( !Options::QuickSaveMode || zCMenu::inGameMenu || isLoading ) return;
 
     oCSavegameInfo* info = gameMan->savegameManager->GetSavegame( iLastSaveSlot );
     if ( !info || !info->DoesSavegameExist() ) return;
 
-    LoadFromMainMenu();
+    if ( Options::LoadLastQuickSaveAfterGameStart || zinput->KeyToggled( Options::KeyQuickLoad ) ) LoadFromMainMenu();
```

Comment:
Move condition before save check.

Reply:
You mean `if ( !info || !info->DoesSavegameExist() ) return;` ?
I'm curious why? When save does not exists it's better to return than try to load even if `LoadLastQuickSaveAfterGameStart ` set to `true`.

Reply:
There is no need to check for the save existence every frame. First, we should check if the game should attempt to load at all.

Reply:
OK, you seems right.